lxd/instance/drivers/driver_qemu.go

```diff
@@ -1487,7 +1487,7 @@ func (d *qemu) Start(stateful bool) error {
 	if ok && cpuLimit != "" {
 		limit, err := strconv.Atoi(cpuLimit)
 		if err == nil {
-			if limit > 1 {
+			if (d.architecture == osarch.ARCH_64BIT_INTEL_X86 || d.architecture == osarch.ARCH_64BIT_S390_BIG_ENDIAN) && limit > 1 {
```

Comment:
I'm not a fan of this sort of duplicated logic because:

1. It makes the code harder to reader (especially as there isn't a comment explaining it).
2. If we want to add support for other architectures in the future, how will we know to add it in both places? And if it only gets added to one of the occurrences then it would introduce bugs. 

Instead the approach I went for when cleaning up UEFI logic was to introduce a small function called `d.architectureSupportsUEFI()` which can be used a anywhere we need to check for it:

https://github.com/lxc/lxd/blob/d4ba0e5ae90f3d6aaa78fa7af58ad35138941379/lxd/instance/drivers/driver_qemu.go#L1665-L1667

This makes the code more readable, and makes it easy to reason about where we need to make changes to support additional architectures.

It also has a the nice side effect of allowing us to see exactly where in the code base these logic hooks are used.

So adding something like `d.architectureSupportsCPUHotplug()` would see to make sense here.